aries_vcx/src/handlers/connection/inviter/state_machine.rs

```diff
@@ -451,15 +450,14 @@ impl SmConnectionInviter {
         new_service_endpoint: String,
     ) -> VcxResult<SignedResponse> {
         request.connection.did_doc.validate()?;
-        let thread_id = &request.get_thread_id().ok_or(VcxError::from_msg(VcxErrorKind::InvalidJson, "Missing ~thread decorator field in request"))?;
-        verify_thread_id(&thread_id, &A2AMessage::ConnectionRequest(request.clone()))?;
+        verify_thread_id(&request.get_thread_id(), &A2AMessage::ConnectionRequest(request.clone()))?;
```

Comment:
Seems like we are validating request thread_id against itself, instead of `self.thread_id`? I think this is verification will always pass

Reply:
Of course, that's a typo.